indexing-service/src/test/java/org/apache/druid/indexing/common/task/concurrent/ActionsTestTask.java

```diff
@@ -49,6 +53,9 @@ public class ActionsTestTask extends CommandQueueTask
 {
   private final TaskActionClient client;
   private final AtomicInteger sequenceId = new AtomicInteger(0);
+  private final Object lock = new Object();
+  @GuardedBy("lock")
+  private final Map<SegmentId, SegmentId> announcedSegmentsToParentSegments = new HashMap<>();
```

Comment:
Why do we need locks here? The only thread writing/reading from this field seems to be the test thread itself.

Reply:
Removed